fe/fe-core/src/main/java/com/starrocks/sql/ast/SetVar.java

```diff
@@ -176,12 +185,15 @@ public void analyze() {
         }
     }
 
-    private void checkNonNegativeLongVariable(String field) {
+    private void checkRangeLongVariable(String field, Long min, Long max) {
         String value = getResolvedExpression().getStringValue();
         try {
             long num = Long.parseLong(value);
-            if (num < 0) {
-                throw new SemanticException(field + " must be equal or greater than 0.");
+            if (min != null && num < min) {
+                throw new SemanticException(String.format("%s must be equal or greater than %d.", field, min));
+            }
+            if (max != null && num > max) {
```

Comment:
why not use Long.MAX_VALUE?

Reply:
Different variables has different uppler limits